app_flutter/lib/task_matrix.dart

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'package:flutter/foundation.dart';
+
+import 'package:cocoon_service/protos.dart' show CommitStatus, Stage, Task;
+
+class TaskMatrix {
+  TaskMatrix({@required this.statuses}) {
+    _columnKeyIndex = _createColumnKeyIndex();
+    _columnMap = <int, int>{};
+    for (int i = 0; i < _columnKeyIndex.length; i++) {
+      _columnMap[i] = i;
+    }
+
+    _matrix = _createTaskMatrix(statuses);
+  }
+
+  final List<CommitStatus> statuses;
+
+  List<Column> _matrix;
+
+  /// A map for taking in the corresponding column in [StatusGrid] and
+  /// where it maps to in [_taskMatrix].
+  Map<int, int> _columnMap;
+
+  /// A key, value table to find what column a [Task] is in.
+  Map<String, int> _columnKeyIndex;
+
+  /// Return [Task] in [_matrix] for a row and col in [StatusGrid].
```

Comment:
dartdoc identifiers have to be in scope - _matrix is private so it shouldn't be mentioned

Reply:
Thank you for pointing that out. Is it okay to phrase it as `"Return [Task] in this matrix for a given row and column in [StatusGrid]"`. This mentions the matrix, but doesn't give the identifier.